snapcraft/commands/lifecycle.py

```diff
@@ -17,16 +17,107 @@
 """Snapcraft lifecycle commands."""
 
 import argparse
+import pathlib
+import textwrap
 from typing import Any
 
 import craft_application.commands
 from craft_cli import emit
 from typing_extensions import override
 
-# pylint: disable=too-many-ancestors
+import snapcraft.pack
 
 
-class SnapCommand(craft_application.commands.lifecycle.PackCommand):
+class PackCommand(craft_application.commands.LifecycleCommand):
+    """Snapcraft pack command."""
+
+    name = "pack"
+    help_msg = "Create the final artifact"
+    overview = textwrap.dedent(
+        """
+        Process parts and create a snap file containing the project payload
+        with the provided metadata. If a directory is specified, pack its
+        contents instead.
+        """
+    )
+
+    @override
+    def _fill_parser(self, parser: argparse.ArgumentParser) -> None:
+        """Add arguments specific to the pack command."""
+        super()._fill_parser(parser)
+
+        parser.add_argument(
+            "directory",
+            metavar="directory",
+            type=str,
+            nargs="?",
+            default=None,
+            help="Directory to pack",
+        )
+        parser.add_argument(
+            "--output",
+            "-o",
+            type=pathlib.Path,
+            default=pathlib.Path(),
+            help="Output directory for created packages",
+        )
+
+    @override
+    def _run(
+        self,
+        parsed_args: argparse.Namespace,
+        step_name: str | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Pack a directory or run the lifecycle and pack all artifacts."""
+        if parsed_args.directory:
+            emit.progress("Packing...")
+            snap_filename = snapcraft.pack.pack_snap(
+                parsed_args.directory, output=str(parsed_args.output)
+            )
+            emit.message(f"Packed {snap_filename}")
+        else:
+            super()._run(parsed_args, step_name="prime")
+            self._services.package.update_project()
```

Comment:
it's unfortunate that this has to be duplicated here, but I don't have any good suggestions

Reply:
I'm glad I'm not the only one who couldn't think of anything better.

The solutions I thought of involve bigger design changes to craft-application and this is not the week for that!